action-scripts/brownie/scripts/script_utils.py

```diff
@@ -121,16 +124,34 @@ def format_into_report(file: dict, transactions: list[dict]) -> str:
     return file_report
 
 
+def prettify_contract_inputs_values(chain, contracts_inputs_values):
+    addr = AddrBook(chain)
+    perm = BalPermissions(chain)
+    outputs = {}
+    for k,v in contracts_inputs_values.items():
+        if "role" in k:
+            v = v.strip('[ ]')
+            v = v.replace(" ", "")
+            v = v.split(",")
+            if len(v) == 1:
+                outputs[k] = f"{v[0]} ({perm.paths_by_action_id.get(v[0], 'N/A')})"
+            else:
+                for value in v:
+                    outputs[k] =[]
```

Comment:
??? Why

Reply:
Not sure I understand the question.  Basically you can have role or roles.  In the case of role you get a single string.  In the case of Roles you get a jsonesque(but not quite) list of addresses.

this basically creates a string that searches for each address in either a single string or a tx_builder_list and replaces it with a similar format that also has the path